src/client.rs

```diff
@@ -97,47 +101,94 @@ impl ActivityPubClient {
     }
 
     pub async fn follow_actor(&self, actor_uri: &str) -> Result<()> {
-        let Actor { id, inbox, .. } = self.get_actor(actor_uri).await?;
         let base = &self.base;
-        info!(%id, %inbox, "sending follow request to inbox");
+        let actor: Actor = self.get_actor(actor_uri).await?;
+        let actor_id = actor.id.as_ref().ok_or(Error::StatusAndMessage {
+            status: StatusCode::BAD_REQUEST,
+            message: "actor has no id",
+        })?;
+        let id = actor_id
```

Comment:
Not for just now, but is it worthwhile having this as a method on the actor itself over in rustypub?

Reply:
i'm not sure how much it would save as we'd want to translate whatever error the library returns to an actiserve error anyway.